### PR TITLE
Kruglov Alexey. Lab 1. Var 4.

### PR DESCRIPTION
**Pull request by KrYackI**
Plugin that renames units(variables, functions, classes) with entered identificator to another entered identificator.

### CODE REVIEW

File: clang/labs/lab1/kruglov_alexey/RenameID.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  explicit RenameVisitor(clang::Rewriter Rewriter, std::string OldName,
+                         std::string NewName)
+      : Rewriter(Rewriter), OldName(OldName), NewName(NewName){};
+
+  bool VisitFunctionDecl(clang::FunctionDecl *Func) {
+    std::string Name = Func->getNameAsString();
+
+    if (Name == OldName) {
+      Rewriter.ReplaceText(Func->getNameInfo().getSourceRange(), NewName);
+      Rewriter.overwriteChangedFiles();
```
Comment (m-ly4) [inline clang/labs/lab1/kruglov_alexey/RenameID.cpp]: May we do this overwrite less frequently?

File: clang/labs/lab1/kruglov_alexey/RenameID.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+class RenameIDPlugin : public clang::PluginASTAction {
+private:
+  std::string OldName;
+  std::string NewName;
+
+protected:
+  bool ParseArgs(const clang::CompilerInstance &CI,
+                 const std::vector<std::string> &args) override {
+    if (args[0].find("OldName=") == std::string::npos || args[1].find("NewName=") == std::string::npos){
```
Comment (m-ly4) [inline clang/labs/lab1/kruglov_alexey/RenameID.cpp]: What if a user passes `SomeOldName` argument?

**Review by m-ly4 — CHANGES_REQUESTED** [review]
Please fix clang-format and clang-tidy remarks if applicable.

File: clang/labs/lab1/kruglov_alexey/RenameID.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+  bool VisitFunctionDecl(clang::FunctionDecl *Func) {
+    std::string Name = Func->getNameAsString();
+
+    if (Name == OldName) {
+      Rewriter.ReplaceText(Func->getNameInfo().getSourceRange(), NewName);
+    }
+    if (Func->getReturnType().getAsString() == OldName ||
+        Func->getReturnType().getAsString() == OldName + " *" ||
+        Func->getReturnType().getAsString() == OldName + " &") {
```
Comment (aobolensk) [inline clang/labs/lab1/kruglov_alexey/RenameID.cpp]: This code is repeated several times across this class. Please, extract it to the separate function

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**